frontend/public/components/environment.jsx

```diff
@@ -80,6 +80,149 @@ const stateToProps = ({k8s}, {obj}) => ({
   model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
 });
 
+class CurrentEnvVars {
+  constructor() {
+    this.currentEnvVars = {};
+  }
+
+  instantiateWithRawData(rawEnvData) {
+    this.isContainerArray = _.isArray(rawEnvData.containers);
+    this.hasInitContainers = !_.isUndefined(rawEnvData.initContainers);
+
+    if (this.isContainerArray) {
+      this.currentEnvVars.containers = envVarsToArray(rawEnvData.containers);
+      this.currentEnvVars.initContainers = envVarsToArray(rawEnvData.initContainers);
+    } else {
+      this.currentEnvVars.buildObject = envVarsToArray(rawEnvData);
+    }
+    return this;
+  }
+
+  instantiateWithResultObject(resultObject, isContainerArray, path) {
```

Comment:
"Result object" is what the editor returns?

Reply:
Maybe `setResultObject`?

Reply:
Same as above where result object is the result of any operations completed by the page, and raw data is the raw data passed in as props.